### ISSUE TITLE
Android: Lack of control of Nested Text locale (Android Span)

### ISSUE THREAD

**Issue by fabriziobertoglio1987** [facebook/react-native#35813]
## Description
Originally part of https://github.com/facebook/react-native/issues/30849

Can not specify the locale (language) of a Nested Text (Android Span). 
TalkBack uses the information when announcing the text.

If the device language is English, TalkBack reads the text in that language.
iOS has the `accessibilityLanguage` property, but there is no equivalent android API.

## React Native version:
main branch

## Expected Behavior
An implementor should be able to specify the type of content within a span of text to give additional context to the TalkBack screenreader (for ex. English, Japanese or Italian).

## Snack
https://snack.expo.dev/ovzKvjHdy

### Version

main

### Output of `npx react-native info`

Not Relevant. It is an issue with ReactAndroid API (main branch)

### Steps to reproduce

- Open the snack
- Enable talkback
- Move the talkback screenreader focus on the Japanese text 

### Snack, code example, screenshot, or link to a repository

https://snack.expo.dev/ovzKvjHdy

**Comment by fabriziobertoglio1987** [comment]
- LocaleSpan: every text in a foreign language needs to be wrapped in a LocaleSpan.
- TextView.textLocale

https://developer.android.com/reference/android/text/style/LocaleSpan
https://youtu.be/x-FcOX6ErdI?t=288
https://github.com/facebook/react-native/pull/35130
